File: swift/Sources/SignalClient/zkgroup/ProfileKeyCommitment.swift
//
// Copyright 2020-2021 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

import Foundation
import SignalFfi

public class ProfileKeyCommitment: ByteArray {

  public static let SIZE: Int = 97

  public required init(contents: [UInt8]) throws {
    try super.init(newContents: contents, expectedLength: ProfileKeyCommitment.SIZE)

    try withUnsafePointerToSerialized { contents in
      try checkError(signal_profile_key_commitment_check_valid_contents(contents))
    }
  }

}

File: java/java/src/main/java/org/signal/zkgroup/profiles/ProfileKeyCommitment.java
//
// Copyright 2020-2021 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

package org.signal.zkgroup.profiles;

import org.signal.zkgroup.InvalidInputException;
import org.signal.zkgroup.internal.ByteArray;
import org.signal.client.internal.Native;

public final class ProfileKeyCommitment extends ByteArray {
  public ProfileKeyCommitment(byte[] contents) throws InvalidInputException {
    super(contents);
    try {
      Native.ProfileKeyCommitment_CheckValidContents(contents);
    } catch (IllegalArgumentException e) {
      throw new InvalidInputException(e.getMessage());
    }
  }
}
